applications/bluesky.md

```diff
@@ -0,0 +1,226 @@
+	- Developer1, former Parity team lead, lecturer at Polkadot Academy, running local Rust meetups, to be finalized
+	- Developer2, Rust / Substrate develdoper, lecturer of Polkadot Academy, to be finalized 
+	- Developer3, TBD
+
+### Contact
+
+- **Contact Name:** Daniel Deng
+- **Contact Email:** gotellme@proton.me 
+- **Website:** 
+
+### Legal Structure
+
+- **Registered Address:** Address of your registered legal entity, if available. Please keep it in a single line. (e.g. High Street 1, London LK1 234, UK)
+- **Registered Legal Entity:** Name of your registered legal entity, if available. (e.g. Duo Ltd.)
+
+### Team's experience
+
+Lecturers of 2022 Polkadot Academy with in-depth domain knowledge and many years of Substrate / Rust software experiences actively developing in Polkadot ecosystem, running local Rust meetups
+
+### Team Code Repos
+
+- https://github.com/layerhero
+
+Please also provide the GitHub accounts of all team members. If they contain no activity, references to projects hosted elsewhere or live are also fine.
+
+- https://github.com/coax1d
+- https://github.com/drahnr
+
+### Team LinkedIn Profiles (if available)
+
+- https://www.linkedin.com/in/dengdaniel
+
+## Development Status 
+
+- references to conversations you might have had related to building this template: convosations in private chats 
+
+## Development Roadmap :nut_and_bolt:
+
+This section should break the development roadmap down into milestones and deliverables. Since these will be part of the agreement, it helps to describe _the functionality we should expect in as much detail as possible_, plus how we can verify and test that functionality. Whenever milestones are delivered, we refer to this document to ensure that everything has been delivered as expected.
+
+### Overview
+
+- **Total Estimated Duration:** Duration of the whole project (e.g. 2 months)
+	- 6 months
+- **Full-Time Equivalent (FTE):**  Average number of full-time employees working on the project throughout its duration (see [Wikipedia](https://en.wikipedia.org/wiki/Full-time_equivalent), e.g. 2 FTE)
+	- 3+ FTE
+- **Total Costs:** Requested amount in USD for the whole project (e.g. 12,000 USD). Note that the acceptance criteria and additional benefits vary depending on the level of funding requested. This and the costs for each milestone need to be provided in USD.
+	- 150,000 USD
+
+### Milestone 1 — Technical showcase example implemented in Ink! Macros and Substrate custom pallets for Ink! / Substrate / Polkadot novice. Publish Article 1/3 
+
+- **Estimated duration:** 2 month
+- **FTE:**  2,3
+- **Costs:** 40,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 1. | Define use case | define a contrived use case that requires 1) business logic 2) reusable components 3) features needing smart contract upgrades 4) performance and low gas fees |
```

Comment:
I would look at OpenZepplin's features like Roles, Ownership (with onlyOwner modifiers, renounceOwnership), etc. These features are used all the time.

Another good topic is String manipulation. ink! provides a `String` feature, but you must opt in to it. It adds a lot of bloat, but is needed for NFT contracts that have Metadata. e.g. calling `tokenURI(tokenId: uint256)` on an ERC721 contract, or `uri(tokenId: uint256)` on ERC1155 usually do some string magic (`baseUri + '/' + tokenId`), but is a pain in ink! and requires code bloat. Perhaps there is a macro or pallet feature you guys could do.

Reply:
Good suggestions. I've incorporated as milestone#1 step2, and milestone#3 step8
https://github.com/dan0ero/Ecosystem-Grants/blob/master/applications/bluesky.md